### ISSUE TITLE
False positive syntax error reporting for comment inside media query 

### ISSUE THREAD

**Issue by ssivanatarajan** [stylelint/stylelint#4274]
> Clearly describe the bug

comment inside a media query cause a syntax error

> Which rule, if any, is the bug related to?

Not a specific rule

> What CSS is needed to reproduce the bug?

e.g.

```
@media only screen and (max-width:520px) {
.freeplan-container p {
	font-size: 18px;
	padding: 8px 0 30px;
}
/* .begin-trial{ margin-top:18px; margin-left: 0;} */
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```not a specific configuration
```

> Which version of stylelint are you using?

e.g. `10.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

e.g. "Node API"

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

e.g. "No"

> What did you expect to happen?

 "No Syntax erros to be flagged."

> What actually happened (e.g. what warnings or errors did you get)?

e.g. "The following warnings were flagged:"

```Got below error message:
CssSyntaxError {
  name: 'CssSyntaxError',
  reason: 'Unknown word',
  source:
   '.freeplan-container p {\n\tfont-size: 18px;\n\tpadding: 8px 0 30px;\n},/* .begin-trial{ margin-top:18px; margin-left: 0;} */',
  line: 4,
  column: 2,
  message: '<css input>:4:2: Unknown word',
  input:
   { line: 4,
     column: 2,
     source:
      '.freeplan-container p {\n\tfont-size: 18px;\n\tpadding: 8px 0 30px;\n},/* .begin-trial{ margin-top:18px; margin-left: 0;} */' } }


```

**Comment by evilebottnawi** [comment]
/cc @stylelint/core We should migrate from `postcss-media-query-parser` because it is abandoned, maybe we can use postcss-value-parser for this case

i can improve parser if we need something

**Comment by jeddy3** [comment]
I can't reproduce this. @ssivanatarajan Can you please provide a reduced configuration to trigger this error?

> We should migrate from `postcss-media-query-parser` because it is abandoned, maybe we can use postcss-value-parser for this case

The parser hasn't proven problematic before and I'm unsure if it's even responsible for this error. Perhaps the most prudent course of action is to keep using it until it fails us and then fork and maintain it?

**Comment by evilebottnawi** [comment]
@jeddy3 we use this plugin in prettier and I find a lot of critical bugs, i can provide them with link, also he use regexp and in some cases it can do DDoS, better migrate from this solution. It is very easy, also we can implement supporting `>=`/`>=`/`>`/`<` in media queries and other good stuff

**Comment by jeddy3** [comment]
> we use this plugin in prettier and I find a lot of critical bugs,

OK. Let's migrate away from it. We might want a new issue for that as we've not sure if this particular `CssSyntaxError` error is connected to that parser.

**Comment by ssivanatarajan** [comment]
@jeddy3  I am able to reproduce this with below rule configuration

'''
{
    "rules": {
        "at-rule-no-unknown": true,
        "block-no-empty": true,
        "color-no-invalid-hex": true

    }
}
'''
Have you checked with Node API?

Note: For the same code snippet i didn't get any error in [https://stylelint.io/demo](stylelint demo)

**Comment by ssivanatarajan** [comment]
This error was thrown from the [cssstats](https://www.npmjs.com/package/cssstats) module. So no issue with style lint.
